### PR TITLE
How to setup mdadm Software RAID

### PR DESCRIPTION
**Pull request by BackInBash**
I have read and understood the Contributor's Certificate of Origin available at the end of
https://raw.githubusercontent.com/hetzneronline/community-content/master/tutorial-template.md
and I hereby certify that I meet the contribution criteria described in it.
Signed-off-by: markus@omg-network.de

### CODE REVIEW

File: tutorials/howto-setup-mdadm/01.de.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    + Bei einem Festplatten Ausfall sind die Daten aller Festplatten verloren.
+
+**RAID1**:
+
+Ist ein Verbund aus zweit oder mehr Partitionen, dabei befinden sich die Daten jeweils gespiegelt auf den Partitionen. 
++ Pro
+    + Erhöht die Ausfallsicherheit / Verfügbarkeit der Daten.
+    + Erhöht die Lesegeschwindigkeit der Daten.
++ Contra
+    + Der Nutzbare Speicherplatz halbiert sich.
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.de.md]:
```suggestion
    + Der verfügbare Speicherplatz halbiert sich.
```

File: tutorials/howto-setup-mdadm/01.en.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    + Higher availability / less storage space
+
+Here is a list of the most used RAID levels:
+
+**RAID0**:
+
+If there is a group of two or more partitions, the partitions are logically combined to one partition.
+Here the availability is reduced. If one of the hard disks is defective ***automatically all data is lost***.
++ Pro
+    + Increases the available storage space.
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.en.md]: You sure? I can increase my physical storage with a RAID0? ;)

File: tutorials/howto-setup-mdadm/01.en.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    + In case of a hard disk failure the data of all hard disks are lost.
+
+**RAID1**:
+
+If there is a group of two or more partitions, the data is mirrored on each partition. 
++ Pro
+    + Increases the reliability / availability of the data.
+    + Increases the reading speed of the data.
++ Contra
+    + The usable storage space is halved.
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.en.md]:
```suggestion
    + The available storage space is halved.
```

File: tutorials/howto-setup-mdadm/01.en.md [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+Mark the newly created partition as a RAID partition:
+```bash
+sudo parted /dev/sdX set 1 raid on
+```
+### Step 2.2 - Create the Software RAID
+Under Linux, `mdadm` is the main tool. It is the interface to the RAID functions of the kernel.
+
+To create a RAID 1 the following command is sufficient:
+```bash
+sudo mdadm --create /dev/md0 --auto md --level=1 --raid-devices=2 /dev/sdX1 /dev/sdX2
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.en.md]: This looks a bit wrong. For me it looks like you have here, e.g. /dev/sda1 and /dev/sda2

You may here use some other letters, as you should add sd<disk1>1 and sd<disk2>1 to md0. That I would also suggest for the other commands where you used sdX. You can also explain your understanding to me if I am wrong.

File: tutorials/howto-setup-mdadm/01.en.md [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+```
+### Step 4.2 - Replace defective hard drive
+To do this, the defective hard disk must first be removed from the RAID:
+```bash
+sudo mdadm --manage /dev/md/X -r /dev/sdX1
+```
+
+If no hot spare hard disk is available, a new hard disk must be partitioned. It is important that the new hard disk has the same partitioning as the defective hard disk!
+
+If the new hard disk is partitioned correctly, it can be added to the RAID system again:
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.en.md]: You may explain here how to copy the partition to the new disk.

Replacing sda if you have two disks, e.g.:
`sfdisk -d /dev/sdb | sfdisk /dev/sda` for MBR
`sgdisk -R=/dev/sda /dev/sdb; sgdisk -G /dev/sda` for GPT

You could also take some things from there: https://wiki.hetzner.de/index.php/Festplattenaustausch_im_Software-RAID/en

File: tutorials/howto-setup-mdadm/01.en.md [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+```
+The progress can be monitored again with the command `watch cat /proc/mdstat`.
+
+Once the rebuild of the RAID is complete, the partition must be removed from the RAID and added again to remove the status "faulty".
+This is done with the commands:
+```bash
+sudo mdadm --manage /dev/md/X -r /dev/sdX1 # To be removed
+sudo mdadm --manage /dev/md/X -a /dev/sdX1 # To be added again
+```
+
```
Comment (n-se) [inline tutorials/howto-setup-mdadm/01.en.md]: You should also update grub and reinstall the disks on grub again as the serial number has changed.

```
update-grub
grub-install /dev/sdX
```

File: tutorials/howto-setup-mdadm/01.en.md (hunk 4 above)
Comment (BackInBash) [reply tutorials/howto-setup-mdadm/01.en.md]: The 'X' is only a space holder.

Comment (n-se) [reply tutorials/howto-setup-mdadm/01.en.md]: I know, but it is kinda confusing if you use the same letter here even if it should be two different disks.

File: tutorials/howto-setup-mdadm/01.en.md (hunk 2 above)
Comment (BackInBash) [reply tutorials/howto-setup-mdadm/01.en.md]: ?

Comment (n-se) [reply tutorials/howto-setup-mdadm/01.en.md]: What I mean is, that you maybe rephrase this sentence. It sounds like I can increase my storage (which is of course not possible)

Maybe "All available storage usable"? Doesn't sound that good... Do you have another idea?

Comment (BackInBash) [reply tutorials/howto-setup-mdadm/01.en.md]: Whats with "Combines the available storage space" or something link that?
